src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx

```diff
@@ -2,19 +2,39 @@ import { PropTypes } from 'prop-types';
 import { useState } from 'react';
 import { Link } from 'react-router-dom';
 import EditShiftHours from '../edit-shift-hours/EditShiftHours';
+import { deleteTimeEntry } from '../../../api/services/timecardService';
+import { UrlSearchParamBuilder } from '../../../utils/api-utils/UrlSearchParamBuilder';
+import { useTimecardContext } from '../../../context/TimecardContext';
+import { deepClone } from '../../../utils/common-utils/common-utils';
 
 const EditShiftTimecard = ({ timeEntry, timeEntriesIndex }) => {
   const toggleEditShiftHours = (event) => {
     event.preventDefault();
     setShowEditShiftHours(!showEditShiftHours);
   };
 
-  const timeEntryIsEmpty =
+  const { timeEntries, setTimeEntries } = useTimecardContext();
+
+  const timeEntryExists =
     'startTime' in timeEntry && timeEntry.startTime !== '';
```

Comment:
a shorter and typed version of this expression would be: `timeEntry?.startTime !== ''`

Reply:
I think when we tried this we had issues because `startTime` doesn't exist in `timeEntry` for new time entries, so it returns true when it should actually return false since it's not a key in the object yet

Reply:
As discussed, the expression I suggested isn't correct. Instead it should have been: 
`const timeEntryExists = timeEntry?.startTime? true : false;`

Or maybe a shorter version to cast to boolean type : `!!timeEntry?.startTime` or `Boolean(timeEntry?.startTime)`